### PR TITLE
SED-1570 ts implement filter bar with multiselect

### PR DESCRIPTION
**Pull request by lucianbaciu**

### CODE REVIEW

File: projects/step-frontend/src/lib/modules/report-nodes/components/call-function-report-node/call-function-report-node.component.html [hunk 1]
```diff
@@ -32,11 +32,7 @@
       <ng-container matColumnDef="actions">
         <th mat-header-cell *matHeaderCellDef></th>
         <td mat-cell *matCellDef="let measure">
-          <a
-            [href]="measure | callFunctionReportNodeExternalLink: node"
-            matTooltip="Search in interactive session"
-            target="_blank"
-          >
+          <a (click)="navigateToAnalyticsView(measure)" matTooltip="Search in interactive session" target="_blank">
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/report-nodes/components/call-function-report-node/call-function-report-node.component.html:35]: `target="_blank"` is redundant here.
When anchor is handled by click, usually we do it in the following way:
```
<a href="#" (click)="navigateToAnalyticsView(measure); false" matTooltip="Search in interactive session">
    <step-icon name="external-link"></step-icon>
</a>
```

File: projects/step-frontend/src/lib/modules/report-nodes/components/call-function-report-node/call-function-report-node.component.ts [hunk 2]
```diff
@@ -17,13 +18,37 @@ export class CallFunctionReportNodeComponent implements OnChanges {
 
   hideMeasures: boolean = false;
 
+  execution: Execution;
+
+  constructor(private executionService: ExecutionStateService) {
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/report-nodes/components/call-function-report-node/call-function-report-node.component.ts:23]: If execution service doesn't use anywhere else except constructor, `private` can be omitted

File: projects/step-frontend/src/lib/modules/timeseries/analytics/analytics-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    this.stopInterval$.next(null);
+    if (newInterval.value) {
+      this.startInterval();
+    }
+  }
+
+  startInterval(delay = 0) {
+    this.stopInterval$.next(null); // make sure to stop it if it's still running
+    timer(delay, this.selectedRefreshInterval.value)
+      .pipe(takeUntil(this.stopInterval$), takeUntil(this.terminator$))
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/analytics/analytics-page.component.ts:134]: If you control your subscription by `stopInternval$`, `terminator$` is redundant
Just don't forget implement ngOnDestroy, emit value to stopInterval$, and complete it

File: projects/step-frontend/src/lib/modules/timeseries/dashboard/time-series-dashboard.component.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        updateRanger: true,
+        updateCharts: true,
+        showLoadingBar: true,
+      })
+      .subscribe();
+  }
+
+  ngOnDestroy(): void {
+    this.terminator$.next(true);
+    this.contextsFactory.destroyContext(this.context.id);
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/dashboard/time-series-dashboard.component.ts:132]: in `ngOnDestroy` Make sure that subscriptions are unsubscribed

File: projects/step-frontend/src/lib/modules/timeseries/dashboard/time-series-dashboard.component.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  @Input() settings!: TimeSeriesDashboardSettings;
+  @ViewChild(PerformanceViewComponent) performanceView!: PerformanceViewComponent;
+
+  timeRangeSelection: TimeRangePickerSelection = { type: RangeSelectionType.FULL };
+  context!: TimeSeriesContext;
+  performanceViewSettings: PerformanceViewSettings | undefined;
+  updateSubscription = new Subscription();
+  queuedSubscription = new Subscription();
+  throttledRefreshTrigger$: Subject<any> = new Subject();
+  terminator$ = new Subject();
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/dashboard/time-series-dashboard.component.ts:32]: If some of those fields above doesn't used in the template, make them private

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/filter-bar/filter-bar.component.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    if (FilterUtils.filterItemIsValid(itemToDelete)) {
+      this.emitFilterChange$.next();
+    }
+  }
+
+  get FilterBarItemType() {
+    return FilterBarItemType;
+  }
+
+  ngOnDestroy(): void {}
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/filter-bar/filter-bar.component.ts:64]: Complete subjects

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/table/timeseries-table.component.ts [hunk 7]
```diff
@@ -23,141 +22,97 @@ export class TimeseriesTableComponent implements OnInit, OnDestroy {
...
-              // TODO update percentiles
-            } else {
-              this.bucketsByKeywords[keywordName] = bucket;
-            }
-            return existingBucketForKeyword;
-          })
-          .sort(this.sortByNameAttributeFn);
-      });
-  }
+  // accumulateData(request: FindBucketsRequest) {
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/table/timeseries-table.component.ts:88]: If it's not required, does it make sense to remove it?

File: projects/step-frontend/src/lib/modules/timeseries/ranger/ts-ranger.component.ts [hunk 8]
```diff
@@ -84,10 +84,12 @@ export class TSRangerComponent implements OnInit, AfterViewInit, OnChanges {
   }
 
   ngOnChanges(changes: SimpleChanges): void {
-    const settings = changes['settings'];
+    let settings = changes['settings'];
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/ranger/ts-ranger.component.ts:87]: Why consts have been changed to `let` here.
If value is immutable inside the code block (method boy, if statement, cycle body, etc) - it should be const.
If you need to reassign it - make it `let`.

File: projects/step-frontend/src/lib/modules/timeseries/time-series-context.ts [hunk 9]
```diff
@@ -8,36 +8,62 @@ import { BucketFilters } from './model/bucket-filters';
...
 
 /**
  * This class is responsible for managing the state of an execution tab. Here we store time selection, colors, filters, etc.
  */
 export class TimeSeriesContext {
-  executionId!: string;
+  id!: string;
   activeExecution: Execution | undefined;
 
+  inProgress$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/time-series-context.ts:21]: Subjects in this class are not completed
Add method that will complete them and invoke it, when you're destroying the context

File: projects/step-frontend/src/lib/modules/timeseries/time-series-context.ts [hunk 10]
```diff
@@ -8,36 +8,62 @@ import { BucketFilters } from './model/bucket-filters';
...
+
   private fullTimeRange: TSTimeRange; // this represents the entire time-series interval. usually this is displayed entirely in the time-ranger
   private readonly fullTimeRangeChange$: Subject<TSTimeRange> = new Subject<TSTimeRange>();
   private selectedTimeRange: TSTimeRange; // this is the zooming selection.
   private readonly selectedTimeRangeChange$: Subject<TSTimeRange> = new Subject<TSTimeRange>();
 
-  activeGroupings: string[] = ['name']; // group dimensions
-  private readonly groupingChangeSubject: Subject<string[]> = new Subject();
+  private readonly activeGroupings: BehaviorSubject<string[]>;
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/time-series-context.ts:28]: Naming. Put $ at the end of name

**Review by dvladir — DISMISSED** [review]

File: projects/step-frontend/src/lib/modules/report-nodes/components/call-function-report-node/call-function-report-node.component.html (hunk 1 above)
Comment (neogucky) [reply projects/step-frontend/src/lib/modules/report-nodes/components/call-function-report-node/call-function-report-node.component.html:35]: We actually agreed to have the ;false "hack" as standard behaviour in FE

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/time-selection/performance-view-time-selection.component.ts [hunk 11]
```diff
@@ -114,13 +114,23 @@ export class PerformanceViewTimeSelectionComponent implements OnInit, OnDestroy
   onRangerSelectionChange(event: TSTimeRange) {
     // check for full range selection
     this.tsContext.updateSelectedRange(event);
+    // if (this.timeLabels[0] === event.from && this.timeLabels[this.timeLabels.length - 1] === event.to) {
+    //
+    // }
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/time-selection/performance-view-time-selection.component.ts:119]: comments

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/time-selection/performance-view-time-selection.component.ts [hunk 12]
```diff
@@ -114,13 +114,23 @@ export class PerformanceViewTimeSelectionComponent implements OnInit, OnDestroy
...
   onRangerZoomReset() {
     this.tsContext.resetZoom();
   }
 
+  resetZoom() {
+    // let selectionToEmit: ExecutionTimeSelection = { type: RangeSelectionType.FULL };
+    // let range = { from: this.rangerSettings!.xValues[0], to: this.rangerSettings!.xValues.slice(-1)[0] };
+    // selectionToEmit.absoluteSelection = range;
+    // this.timeSelectionState.resetZoom(range);
+  }
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/time-selection/performance-view-time-selection.component.ts:132]: comments

File: projects/step-frontend/src/lib/modules/timeseries/analytics/analytics-page.component.ts (hunk 3 above)
Comment (lucianbaciu) [reply projects/step-frontend/src/lib/modules/timeseries/analytics/analytics-page.component.ts:134]: stopInterval is just for this particular interval, while terminator$ is the global destroy trigger. I think having both is better here.

**Review by neogucky — APPROVED** [review]

**Pull request merged**